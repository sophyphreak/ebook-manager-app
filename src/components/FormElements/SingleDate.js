import React, { Component } from 'react';
import BasicInput from './MainElements/BasicInput';
import { SingleDatePicker } from 'react-dates';
import { FormGroup, Label } from 'reactstrap';

const SingleDate = ({
  label,
  date,
  onDateChange,
  calendarFocused,
  onFocusChange
}) => (
  <FormGroup>
    <Label>{label}</Label>
    <br />
    <SingleDatePicker
      date={date}
      onDateChange={onDateChange}
      focused={calendarFocused}
      onFocusChange={onFocusChange}
    />
  </FormGroup>
);

export default SingleDate;
