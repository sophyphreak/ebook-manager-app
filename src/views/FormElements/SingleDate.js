import React from 'react';
import { SingleDatePicker } from 'react-dates';
import { FormGroup, Label } from 'reactstrap';

const SingleDate = ({
  label,
  date,
  onDateChange,
  calendarFocused,
  onFocusChange
}) => (
  <FormGroup>
    <Label>{label}</Label>
    <br />
    <SingleDatePicker
      date={date}
      onDateChange={onDateChange}
      focused={calendarFocused}
      onFocusChange={onFocusChange}
    />
  </FormGroup>
);

export default SingleDate;
