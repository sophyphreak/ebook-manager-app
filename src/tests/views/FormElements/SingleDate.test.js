import React from 'react';
import { shallow } from 'enzyme';
import SingleDate from '../../../views/FormElements/SingleDate';

let onDateChange, onFocusChange, wrapper;

beforeEach(() => {
  onDateChange = jest.fn();
  onFocusChange = jest.fn();
  wrapper = shallow(
    <SingleDate onDateChange={onDateChange} onFocusChange={onFocusChange} />
  );
});

test('should render SingleDate correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
