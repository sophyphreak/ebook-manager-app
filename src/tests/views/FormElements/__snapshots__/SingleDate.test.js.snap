// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SingleDate correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleDate
    onDateChange={[MockFunction]}
    onFocusChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        />,
        <br />,
        <withStyles(SingleDatePicker)
          anchorDirection="left"
          appendToBody={false}
          block={false}
          calendarInfoPosition="bottom"
          customCloseIcon={null}
          customInputIcon={null}
          date={null}
          daySize={39}
          disableScroll={false}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          horizontalMonthPadding={13}
          id="date"
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepFocusOnInput={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          noBorder={false}
          numberOfMonths={2}
          onClose={[Function]}
          onDateChange={[MockFunction]}
          onFocusChange={[MockFunction]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          openDirection="down"
          orientation="horizontal"
          phrases={
            Object {
              "calendarLabel": "Calendar",
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsSelected": [Function],
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month.",
              "jumpToPrevMonth": "Move backward to switch to the previous month.",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
              "moveFocusByOneMonth": "Switch months.",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
              "openThisPanel": "Open this panel.",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field.",
              "selectFocusedDate": "Select the date in focus.",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          regular={false}
          renderCalendarInfo={null}
          renderDayContents={null}
          renderMonthElement={null}
          renderMonthText={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          small={false}
          verticalHeight={null}
          verticalSpacing={22}
          weekDayFormat="dd"
          withFullScreenPortal={false}
          withPortal={false}
        />,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
          "tag": "label",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorDirection": "left",
          "appendToBody": false,
          "block": false,
          "calendarInfoPosition": "bottom",
          "customCloseIcon": null,
          "customInputIcon": null,
          "date": null,
          "dayAriaLabelFormat": undefined,
          "daySize": 39,
          "disableScroll": false,
          "disabled": false,
          "displayFormat": [Function],
          "enableOutsideDays": false,
          "firstDayOfWeek": null,
          "focused": false,
          "hideKeyboardShortcutsPanel": false,
          "horizontalMargin": 0,
          "horizontalMonthPadding": 13,
          "id": "date",
          "initialVisibleMonth": null,
          "inputIconPosition": "before",
          "isDayBlocked": [Function],
          "isDayHighlighted": [Function],
          "isOutsideRange": [Function],
          "isRTL": false,
          "keepFocusOnInput": false,
          "keepOpenOnDateSelect": false,
          "monthFormat": "MMMM YYYY",
          "navNext": null,
          "navPrev": null,
          "noBorder": false,
          "numberOfMonths": 2,
          "onClose": [Function],
          "onDateChange": [MockFunction],
          "onFocusChange": [MockFunction],
          "onNextMonthClick": [Function],
          "onPrevMonthClick": [Function],
          "openDirection": "down",
          "orientation": "horizontal",
          "phrases": Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          },
          "placeholder": "Date",
          "readOnly": false,
          "regular": false,
          "renderCalendarDay": undefined,
          "renderCalendarInfo": null,
          "renderDayContents": null,
          "renderMonthElement": null,
          "renderMonthText": null,
          "reopenPickerOnClearDate": false,
          "required": false,
          "screenReaderInputMessage": "",
          "showClearDate": false,
          "showDefaultInputIcon": false,
          "small": false,
          "transitionDuration": undefined,
          "verticalHeight": null,
          "verticalSpacing": 22,
          "weekDayFormat": "dd",
          "withFullScreenPortal": false,
          "withPortal": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          />,
          <br />,
          <withStyles(SingleDatePicker)
            anchorDirection="left"
            appendToBody={false}
            block={false}
            calendarInfoPosition="bottom"
            customCloseIcon={null}
            customInputIcon={null}
            date={null}
            daySize={39}
            disableScroll={false}
            disabled={false}
            displayFormat={[Function]}
            enableOutsideDays={false}
            firstDayOfWeek={null}
            focused={false}
            hideKeyboardShortcutsPanel={false}
            horizontalMargin={0}
            horizontalMonthPadding={13}
            id="date"
            initialVisibleMonth={null}
            inputIconPosition="before"
            isDayBlocked={[Function]}
            isDayHighlighted={[Function]}
            isOutsideRange={[Function]}
            isRTL={false}
            keepFocusOnInput={false}
            keepOpenOnDateSelect={false}
            monthFormat="MMMM YYYY"
            navNext={null}
            navPrev={null}
            noBorder={false}
            numberOfMonths={2}
            onClose={[Function]}
            onDateChange={[MockFunction]}
            onFocusChange={[MockFunction]}
            onNextMonthClick={[Function]}
            onPrevMonthClick={[Function]}
            openDirection="down"
            orientation="horizontal"
            phrases={
              Object {
                "calendarLabel": "Calendar",
                "chooseAvailableDate": [Function],
                "clearDate": "Clear Date",
                "closeDatePicker": "Close",
                "dateIsSelected": [Function],
                "dateIsUnavailable": [Function],
                "enterKey": "Enter key",
                "escape": "Escape key",
                "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
                "homeEnd": "Home and end keys",
                "jumpToNextMonth": "Move forward to switch to the next month.",
                "jumpToPrevMonth": "Move backward to switch to the previous month.",
                "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
                "keyboardShortcuts": "Keyboard Shortcuts",
                "leftArrowRightArrow": "Right and left arrow keys",
                "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
                "moveFocusByOneMonth": "Switch months.",
                "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
                "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
                "openThisPanel": "Open this panel.",
                "pageUpPageDown": "page up and page down keys",
                "questionMark": "Question mark",
                "returnFocusToInput": "Return to the date input field.",
                "selectFocusedDate": "Select the date in focus.",
                "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
                "upArrowDownArrow": "up and down arrow keys",
              }
            }
            placeholder="Date"
            readOnly={false}
            regular={false}
            renderCalendarInfo={null}
            renderDayContents={null}
            renderMonthElement={null}
            renderMonthText={null}
            reopenPickerOnClearDate={false}
            required={false}
            screenReaderInputMessage=""
            showClearDate={false}
            showDefaultInputIcon={false}
            small={false}
            verticalHeight={null}
            verticalSpacing={22}
            weekDayFormat="dd"
            withFullScreenPortal={false}
            withPortal={false}
          />,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "tag": "label",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorDirection": "left",
            "appendToBody": false,
            "block": false,
            "calendarInfoPosition": "bottom",
            "customCloseIcon": null,
            "customInputIcon": null,
            "date": null,
            "dayAriaLabelFormat": undefined,
            "daySize": 39,
            "disableScroll": false,
            "disabled": false,
            "displayFormat": [Function],
            "enableOutsideDays": false,
            "firstDayOfWeek": null,
            "focused": false,
            "hideKeyboardShortcutsPanel": false,
            "horizontalMargin": 0,
            "horizontalMonthPadding": 13,
            "id": "date",
            "initialVisibleMonth": null,
            "inputIconPosition": "before",
            "isDayBlocked": [Function],
            "isDayHighlighted": [Function],
            "isOutsideRange": [Function],
            "isRTL": false,
            "keepFocusOnInput": false,
            "keepOpenOnDateSelect": false,
            "monthFormat": "MMMM YYYY",
            "navNext": null,
            "navPrev": null,
            "noBorder": false,
            "numberOfMonths": 2,
            "onClose": [Function],
            "onDateChange": [MockFunction],
            "onFocusChange": [MockFunction],
            "onNextMonthClick": [Function],
            "onPrevMonthClick": [Function],
            "openDirection": "down",
            "orientation": "horizontal",
            "phrases": Object {
              "calendarLabel": "Calendar",
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsSelected": [Function],
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month.",
              "jumpToPrevMonth": "Move backward to switch to the previous month.",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
              "moveFocusByOneMonth": "Switch months.",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
              "openThisPanel": "Open this panel.",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field.",
              "selectFocusedDate": "Select the date in focus.",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
              "upArrowDownArrow": "up and down arrow keys",
            },
            "placeholder": "Date",
            "readOnly": false,
            "regular": false,
            "renderCalendarDay": undefined,
            "renderCalendarInfo": null,
            "renderDayContents": null,
            "renderMonthElement": null,
            "renderMonthText": null,
            "reopenPickerOnClearDate": false,
            "required": false,
            "screenReaderInputMessage": "",
            "showClearDate": false,
            "showDefaultInputIcon": false,
            "small": false,
            "transitionDuration": undefined,
            "verticalHeight": null,
            "verticalSpacing": 22,
            "weekDayFormat": "dd",
            "withFullScreenPortal": false,
            "withPortal": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
